celestia/src/common.rs

```diff
@@ -0,0 +1,126 @@
+use std::net::SocketAddr;
+use std::path::PathBuf;
+
+use anyhow::{bail, Result};
+use celestia_node::network::Network;
+use clap::{error::ErrorKind, ArgGroup, CommandFactory, Parser, ValueEnum};
+use libp2p::multiaddr::Protocol;
+use libp2p::Multiaddr;
+use serde_repr::{Deserialize_repr, Serialize_repr};
+
+use crate::{native, server};
+
+const SERVER_DEFAULT_BIND_ADDR: &str = "127.0.0.1:9876";
+
+#[derive(Debug, Parser)]
+// disallow specifying store path when running in browser
+#[clap(group(ArgGroup::new("native_xor_browser")
+             .args(&["store", "browser"])))]
+pub(crate) struct Args {
+    /// Network to connect.
+    #[arg(short, long, value_enum, default_value_t)]
+    pub(crate) network: ArgNetwork,
+
+    /// Listening addresses. Can be used multiple times.
+    #[arg(short, long = "listen")]
+    pub(crate) listen_addrs: Vec<Multiaddr>,
+
+    /// Bootnode multiaddr, including peer id. Can be used multiple times.
+    #[arg(short, long = "bootnode")]
+    pub(crate) bootnodes: Vec<Multiaddr>,
+
+    /// Persistent header store path.
+    #[arg(short, long = "store")]
+    pub(crate) store: Option<PathBuf>,
+
+    /// Serve wasm node which can be accessed with web browser
+    #[arg(long)]
+    browser: bool,
+}
+
+#[derive(
+    Debug, Default, Clone, Copy, PartialEq, Eq, ValueEnum, Serialize_repr, Deserialize_repr,
+)]
+#[repr(u8)]
```

Comment:
why do we need `repr(u8)` there now and `Serialize_repr`, `Deserialize_repr`? Previously it was working fine with
```rs
#[derive(Debug, Default, Clone, Copy, PartialEq, Eq, ValueEnum)]
enum Network {
    Arabica,
    Mocha,
    #[default]
    Private,
}
```

Reply:
Serialize is used by axum as this field is returned as part of the configuration in `WasmNodeArgs` returned as JSON, deleted Deserialize as it wasn't needed.
`repr(u8)` is required by `Serialize_repr. I'm using that because it matches how wasm-bindgen represents rust enums in js, as numbers, so I can use them directly without conversion.